import-scripts/fetch-ddp-and-import-cmo-access-data.sh

```diff
@@ -0,0 +1,262 @@
+        old_dir=$(pwd)
+        if ! cd "$GIT_REPO_HOME" ; then
+            echo "fetch_git_repo_data_updates() : error - could not change to directory $GIT_REPO_HOME"
+            return 1
+        fi
+        # drop local changes to files
+        if ! git checkout . ; then
+            echo "fetch_git_repo_data_updates() : error - could not checkout default branch at $GIT_REPO_HOME"
+            return 1
+        fi
+        # clean local files and directories which are not part of the repository
+        if ! git clean -fd ; then
+            echo "fetch_git_repo_data_updates() : error - could not git clean at $GIT_REPO_HOME"
+            return 1
+        fi
+        # lose any unpushed local commits
+        if ! git reset --hard "origin/$GIT_BRANCH_NAME" ; then
+            echo "fetch_git_repo_data_updates() : error - could not git reset --hard origin/$GIT_BRANCH_NAME at $GIT_REPO_HOME"
+            return 1
+        fi
+        # pull updates from repo
+        git_pull_stdout_filepath=$(mktemp "$CMO_ACCESS_TMPDIR/git_pull.stdout_XXXXXX")
+        if ! git pull origin "$GIT_BRANCH_NAME" > "$git_pull_stdout_filepath" ; then
+            echo "fetch_git_repo_data_updates() : error - could not git pull origin $GIT_BRANCH_NAME at $GIT_REPO_HOME"
+            return 1
+        fi
+        # always run git-lfs pull origin <branch> to ensure git-lfs managed file contents show actual data, not pointer file reference info
+        if ! git-lfs pull origin "$GIT_BRANCH_NAME" ; then
+            echo "fetch_git_repo_data_updates() : error - could not git-lfs pull origin $GIT_BRANCH_NAME at $GIT_REPO_HOME"
+            return 1
+        fi
+        # updates might have been fetched already but still want to notify if updates were pulled or not
+        if grep "Already up to date." "$git_pull_stdout_filepath" ; then
+            echo "fetch_git_repo_data_updates() : we have the latest dataset at $GIT_REPO_HOME"
+        else
+            echo "fetch_git_repo_data_updates() : we pulled updates from git at $GIT_REPO_HOME"
+        fi
+        rm "$git_pull_stdout_filepath"
+        cd "$old_dir"
+        return 0
+    }
+
+    function fetch_ddp_demographics_and_timeline() {
+        rm -f "$cmo_access_dmp_pids_filepath"
+        rm -f "$cmo_access_seq_date_filepath"
+        rm -rf "$cmo_access_ddp_output_dirpath"
+        mkdir "$cmo_access_ddp_output_dirpath"
+        echo "SAMPLE_ID	PATIENT_ID	SEQ_DATE" > "$cmo_access_seq_date_filepath" # create placeholder seq_date file
+        egrep -v "^#" $CMO_ACCESS_DATA_HOME/data_clinical_patient.txt | awk -F'\t' 'NR==1 { for (i=1; i<=NF; i++) { f[$i] = i } }{ if ($f["PATIENT_ID"] != "PATIENT_ID") { print $(f["PATIENT_ID"]) } }' | egrep "^P-" | sort | uniq > $cmo_access_dmp_pids_filepath
+        CMO_ACCESS_DDP_DEMOGRAPHICS_RECORD_COUNT=$(wc -l < $cmo_access_dmp_pids_filepath)
+        if [ $CMO_ACCESS_DDP_DEMOGRAPHICS_RECORD_COUNT -le $DEFAULT_DDP_DEMOGRAPHICS_ROW_COUNT ] ; then
```

Comment:
where is DEFAULT_DDP_DEMOGRAPHICS_ROW_COUNT defined? is this a check to make sure we don't have no records?

Reply:
This environment variable is defined in dmp-import-vars-functions.sh as "2". I think there is some error which occurs if you query for just a single ddp fetch, so we always want a minimum of 2 patients when doing a ddp fetch. I don't think I have made any modifications to dmp-import-vars-functions.sh.